### ISSUE TITLE
GCop 515

### ISSUE THREAD

**Issue by marjanjavid** [Geeksltd/GCop#165]
Please check this rule. It seems this rule is not kicked in an expected place.

**Comment by Foroughi** [comment]
@marjanjavid Would you please provide an example or be more specific?

**Comment by marjanjavid** [comment]
of course. but first of all i should say i am not important that in my sample this line of code is considered or not 
"if (!methodInfo.ContainingType?.BaseType?.BaseType?.ToString().StartsWith("MSharp.Framework.Entity") ?? true) return;" because i don't have enough information about MSharp.

there is a rule in M# about Unique properties. If we emphasize a property as a unique one, like "String("Email", 100).Mandatory().Unique().Accepts(TextPattern.EmailAddress);" then it will automatically generate a static method.this method is  FindByPropertyName, where PropertyName is the name of your unique property.

So if there is a method which starts with "FindBy" keyword followed by the name of Not unique property , this GCop error should be kicked.

so i wrote a method with this features for User, but i am not sure about the rule about MSharp.
